return-result-to-caller-test/src/main/java/com/hazelcast/jet/tests/returnresult/ReturnResultToCallerTest.java

```diff
@@ -0,0 +1,188 @@
+import static com.hazelcast.jet.core.JobStatus.RUNNING;
+import static com.hazelcast.jet.pipeline.WindowDefinition.tumbling;
+import static com.hazelcast.jet.tests.common.Util.entry;
+import static com.hazelcast.jet.tests.common.Util.getJobStatusWithRetry;
+import static com.hazelcast.jet.tests.common.Util.sleepMinutes;
+import static com.hazelcast.jet.tests.common.Util.waitForJobStatus;
+
+public class ReturnResultToCallerTest extends AbstractSoakTest {
+
+    private static final int DEFAULT_ITEMS_PER_SECOND = 100;
+    private static final int DEFAULT_WINDOW_SIZE = 100;
+    private static final int DEFAULT_ALLOWED_LAG = 1000;
+    private static final int SECONDS_IN_MINUTE = 60;
+
+    private static final String OBSERVABLE_NAME = ReturnResultToCallerTest.class.getSimpleName();
+
+    private static int itemsPerSecond;
+    private static int itemsPerMinute;
+    private static int windowSize;
+    private static int allowedLag;
+
+    private Observable<Map.Entry<Long, Long>> observable;
+    private TestObserver observer;
+
+    public static void main(String[] args) throws Exception {
+        new ReturnResultToCallerTest().run(args);
+    }
+
+    @Override
+    public void init() {
+        itemsPerSecond = propertyInt("itemsPerSecond", DEFAULT_ITEMS_PER_SECOND);
+        itemsPerMinute = itemsPerSecond * SECONDS_IN_MINUTE;
+        windowSize = propertyInt("windowSize", DEFAULT_WINDOW_SIZE);
+        allowedLag = propertyInt("allowedLag", DEFAULT_ALLOWED_LAG);
+
+        observable = jet.getObservable(OBSERVABLE_NAME);
+        observer = new TestObserver(logger);
+        observable.addObserver(observer);
+    }
+
+    @Override
+    public void test() throws Throwable {
+        JobConfig jobConfig = new JobConfig()
+                .setName("ReturnResultToCallerTest");
+        Job job = jet.newJob(pipeline(), jobConfig);
+        waitForJobStatus(job, RUNNING);
+        sleepMinutes(1);
+
+        int lastTs = -1;
+
+        long begin = System.currentTimeMillis();
+        while (System.currentTimeMillis() - begin < durationInMillis) {
+            JobStatus jobStatus = getJobStatusWithRetry(job);
+            assertEquals(RUNNING, jobStatus);
+
+            Throwable onNextError = observer.getOnNextError();
+            if (onNextError != null) {
+                throw onNextError;
+            }
+
+            Throwable error = observer.getError();
+            if (error != null) {
+                throw error;
+            }
+            assertEquals(0, observer.getCompletions());
+
+            int newLastTs = observer.getLastTs();
+            assertTrue("Observer did not see any new item. Last viewed item: " + lastTs, newLastTs > lastTs);
+
+            lastTs = newLastTs;
+
+            sleepMinutes(1);
+        }
+
+        job.cancel();
+    }
+
+    private Pipeline pipeline() {
+
+        Pipeline p = Pipeline.create();
+        p.readFrom(TestSources.itemStream(itemsPerSecond))
```

Comment:
the source runs on single member and rest of the pipeline does not force it to shuffle. So this job will run on a single member, do we want that ?